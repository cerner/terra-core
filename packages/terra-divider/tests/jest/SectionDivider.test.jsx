import React from 'react';
import ThemeContextProvider from 'terra-theme-context/lib/ThemeContextProvider';

import SectionDivider from '../../src/_SectionDivider';

describe('Divider', () => {
  const defaultRender = <SectionDivider />;

  // Snapshot Tests
  it('should render a basic component', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper).toMatchSnapshot();
  });

  it('it should pass the All the Props', () => {
    const wrapper = shallow(<SectionDivider id="testDivider" text="custom-text" level={3} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should apply the theme context className to the text divider', () => {
    const wrapper = mount(
      <ThemeContextProvider theme={{ className: 'theme-class-name' }}>
        <SectionDivider text="custom-text" />
      </ThemeContextProvider>,
    );
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find('div').hasClass('theme-class-name')).toBe(true);
  });
});
