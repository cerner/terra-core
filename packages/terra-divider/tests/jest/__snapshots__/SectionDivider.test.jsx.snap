// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider it should pass the All the Props 1`] = `
<div
  className="divider-container"
>
  <h3
    className="divider-text"
    id="testDivider"
  >
    custom-text
  </h3>
</div>
`;

exports[`Divider should apply the theme context className to the text divider 1`] = `
<ThemeContextProvider
  theme={
    Object {
      "className": "theme-class-name",
    }
  }
>
  <SectionDivider
    id="testDivider"
    level={3}
    text="custom-text"
  >
    <div
      className="divider-container theme-class-name"
    >
      <h3
        className="divider-text"
        id="testDivider"
      >
        custom-text
      </h3>
    </div>
  </SectionDivider>
</ThemeContextProvider>
`;

exports[`Divider should render a basic component 1`] = `
<div
  className="divider-container"
/>
`;
